content/Development/qtest.md

```diff
@@ -194,3 +194,15 @@ Remote debugging with Query File Test is a potent tool for debugging Hive. With
 ```sh
 $ mvn -Pitests -pl itests/qtest -Dmaven.surefire.debug test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=alter1.q
 ```
+
+## Tips for Adding New Tests in Hive
+
+Following are a few rules of thumb that should be followed when adding new test cases in Hive that require the introduction of new query file(s). Of course, these rules should not be applied if they invalidate the purpose of your test to begin with. These are generally helpful in keeping the test queries concise, minimizing the redundancies where possible, and ensuring that cascading failures due to a single test failure do not occur.
+
+* Instead of creating your own data file for loading into a new table, use existing data from staged tables like `src`.
```

Comment:
nit: I like tests that are self-contained and minimal so I am not sure if using built-in datasets (especially `src` where the data are pretty cryptic) is always a good idea. Anyways this PR just moves around existing content so I am fine if it remains as is.

Reply:
I'm keeping this statement because we might have not achieved consensus.